client/header/activity-panel/panels/inbox/index.js

```diff
@@ -66,32 +47,61 @@ class InboxPanel extends Component {
 		);
 	}
 
-	renderNotes() {
-		const { lastRead, notes } = this.props;
+	renderNotes( hasNotes ) {
+		const { isUndoRequesting, lastRead, notes } = this.props;
 
-		const validNotes = filter( notes, ( note ) => {
-			const { is_deleted: isDeleted } = note;
-			const noteActive = has( note, 'is_deleted' ) ? ! isDeleted : true;
-			return noteActive;
-		} );
+		if ( isUndoRequesting === 'undo-dismiss-all' ) {
+			return;
+		}
 
-		if ( validNotes.length === 0 ) {
+		if ( ! hasNotes ) {
 			return this.renderEmptyCard();
 		}
 
 		const notesArray = Object.keys( notes ).map( ( key ) => notes[ key ] );
 
-		return notesArray.map( ( note ) => (
-			<InboxNoteCard
-				key={ note.id }
-				note={ note }
-				lastRead={ lastRead }
-			/>
-		) );
+		return notesArray.map( ( note ) => {
+			const showPlaceholder = isUndoRequesting
+				? isUndoRequesting === note.id
+				: false;
+			if ( showPlaceholder ) {
+				return (
+					<InboxNotePlaceholder
+						className={ 'banner message-is-unread' }
```

Comment:
Nitpick: this could just use a double quoted attribute instead braces.

Reply:
It was modified in the commit [1c66636](https://github.com/woocommerce/woocommerce-admin/pull/4379/commits/1c66636a4a84075b57f9c3d33ade74f25075d57c)